# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/filedialog/WinFileDialogParent.h
# Commit: d56484304ec3
# Full Hash: d56484304ec3abe3891ede9e8d141c95543d3e37
# Author: Ray Kraesig <rkraesig@mozilla.com>
# Date: 2023-09-16 09:14:45
# Regressor Bug: 1837079
# File Overlap Count: 2
# Description:
#   Bug 1837079 - [5/10] Create WinFileDialog IPDL protocol and implementation  r=gstoll,handyman,ipc-reviewers,nika
#   
#   Create and implement a new top-level IPC protocol, `PWinFileDialog`,
#   using the primitives from bug 1833450 and the new sandboxing type from
#   the previous commit.
# ==============================================================================

diff -r 36d3afd881b4 -r d56484304ec3 widget/windows/filedialog/WinFileDialogParent.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/widget/windows/filedialog/WinFileDialogParent.h	Fri Sep 15 18:10:01 2023 +0000
@@ -0,0 +1,43 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef widget_windows_filedialog_WinFileDialogParent_h__
+#define widget_windows_filedialog_WinFileDialogParent_h__
+
+#include "mozilla/Assertions.h"
+#include "mozilla/Maybe.h"
+#include "mozilla/MozPromise.h"
+#include "mozilla/ProcInfo.h"
+#include "mozilla/Result.h"
+#include "mozilla/SpinEventLoopUntil.h"
+#include "mozilla/dom/ChromeUtilsBinding.h"
+#include "mozilla/ipc/UtilityProcessParent.h"
+#include "mozilla/widget/filedialog/PWinFileDialogParent.h"
+#include "nsISupports.h"
+#include "nsStringFwd.h"
+
+namespace mozilla::widget::filedialog {
+
+class WinFileDialogParent : public PWinFileDialogParent {
+ public:
+  using UtilityActorName = ::mozilla::UtilityActorName;
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(WinFileDialogParent, override);
+
+ public:
+  WinFileDialogParent();
+  nsresult BindToUtilityProcess(
+      mozilla::ipc::UtilityProcessParent* aUtilityParent);
+
+  UtilityActorName GetActorName() {
+    return UtilityActorName::WindowsFileDialog;
+  }
+
+ private:
+  ~WinFileDialogParent();
+};
+}  // namespace mozilla::widget::filedialog
+
+#endif  // widget_windows_filedialog_WinFileDialogParent_h__